google-cloud-storage/test/google/cloud/storage/retry/utils.rb

```diff
@@ -0,0 +1,462 @@
+
+  def self.test_permissions client, _preconditions, **resources
+    bucket = resources[:bucket]
+    permissions = ["storage.buckets.get", "storage.buckets.create"]
+    bucket.test_permissions permissions
+  end
+
+  def self.list_default_acls client, _preconditions, **resources
+    bucket_name = resources[:bucket].name
+    client.list_default_acls bucket_name
+  end
+
+  def self.delete_hmac_key client, _preconditions, **resources
+    access_id = resources[:hmac_key].access_id
+    client.delete_hmac_key access_id
+  end
+
+  def self.get_hmac_key client, _preconditions, **resources
+    access_id = resources[:hmac_key].access_id
+    client.get_hmac_key access_id
+  end
+
+  def self.hmac_key_reload client, _preconditions, **resources
+    hmac_key = resources[:hmac_key]
+    hmac_key.reload!
+  end
+
+  def self.list_hmac_keys client, _preconditions, **resources
+    hmac_keys = client.list_hmac_keys
+    hmac_keys.items.each do |hmac_key|
+      next
+    end
+  end
+
+  def self.delete_notification client, _preconditions, **resources
+    bucket_name = resources[:bucket].name
+    notification_id = resources[:notification].id
+    client.delete_notification bucket_name, notification_id
+  end
+
+  def self.get_notification client, _preconditions, **resources
+    bucket_name = resources[:bucket].name
+    notification_id = resources[:notification].id
+    client.get_notification bucket_name, notification_id
+  end
+
+  def self.bucket_get_notification client, _preconditions, **resources
+    bucket = resources[:bucket]
+    notification_id = resources[:notification].id
+    bucket.notification notification_id
+  end
+
+  def self.list_notifications client, _preconditions, **resources
+    bucket_name = resources[:bucket].name
+    notifications = client.list_notifications bucket_name
+    notifications.items.each do |notification|
+      next
+    end
+  end
+
+  def self.bucket_notifications client, _preconditions, **resources
+    bucket = resources[:bucket]
+    notifications = bucket.notifications
+    notifications.each do |notification|
+      next
+    end
+  end
+
+  def self.list_file_acls client, _preconditions, **resources
+    bucket_name = resources[:bucket].name
+    object_name = resources[:object].name
+    client.list_file_acls bucket_name, object_name
+  end
+
+  def self.project_service_account client, _preconditions, **resources
+    client.project_service_account
+  end
+
+  def self.patch_bucket client, _preconditions, **resources
+    bucket = resources[:bucket]
+    metageneration = resources[:bucket].metageneration
+    bucket_gapi = Google::Apis::StorageV1::Bucket.new storage_class: "COLDLINE"
+    if _preconditions
+      client.patch_bucket bucket.name, bucket_gapi, if_metageneration_match: metageneration
+    else
+      client.patch_bucket bucket.name, bucket_gapi
+    end
+  end
+
+  def self.bucket_acl_public client, _preconditions, **resources
+    bucket = resources[:bucket]
+    if _preconditions
```

Comment:
Typically _varname is used when the variable is not used. Might want to change the variable name here.

Reply:
+1